File: app/src/main/java/com/savvasdalkitsis/gameframe/widget/view/PowerWidgetProvider.java
package com.savvasdalkitsis.gameframe.widget.view;

import com.savvasdalkitsis.gameframe.R;

public class PowerWidgetProvider extends ClickableWidgetProvider {

    @Override
    protected int layoutResId() {
        return R.layout.widget_power;
    }

    @Override
    protected void onClick() {
        presenter.power();
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/feature/widget/view/PowerWidgetProvider.kt
package com.savvasdalkitsis.gameframe.feature.widget.view

import com.savvasdalkitsis.gameframe.R

class PowerWidgetProvider : ClickableWidgetProvider() {

    override fun layoutResId() = R.layout.widget_power

    override fun onClick() {
        presenter.power()
    }
}
